CsharpExtras/Map/Dictionary/Curry/ICurryDictionary.cs

```diff
@@ -8,6 +8,8 @@ public interface ICurryDictionary<TKey, TVal>
         NonnegativeInteger Arity { get; }
         TVal this[params TKey[] keyTuple] { get; }
         IEnumerable<IList<TKey>> Keyset();
+        IEnumerable<(IList<TKey>, TVal)> KeyValuePairs();
+        IEnumerable<TVal> Values();
```

Comment:
Should these be methods or properties? The standard C# dictionaries expose `Keys` and `Values` as properties. Is there a particular reason you made these methods instead?

Reply:
Yeah I originally had Keys as a property, but it felt like a lie to the user as the enumeration was walking through each individual item... However, I suppose that's the way for most enumerable properties... I'll change em to properties so